src/main/cpp/subsystems/Shooter.cpp

```diff
@@ -0,0 +1,52 @@
+#include "subsystems/Shooter.h"
+
+Shooter::Shooter()
+: m_firing_motorL(1, rev::CANSparkMaxLowLevel::MotorType::kBrushless),
+  m_firing_motorR(2, rev::CANSparkMaxLowLevel::MotorType::kBrushless),
+  m_arm_motorT(14, rev::CANSparkMaxLowLevel::MotorType::kBrushless), 
+  m_arm_motorB(5, rev::CANSparkMaxLowLevel::MotorType::kBrushless),
+  m_intake(3),
+  m_arm_PID(0.1, 0.0, 0.1),
+  m_arm_encoderT(m_arm_motorT.GetAbsoluteEncoder(rev::SparkMaxAbsoluteEncoder::Type::kDutyCycle)),
+  m_arm_encoderB(m_arm_motorB.GetAbsoluteEncoder(rev::SparkMaxAbsoluteEncoder::Type::kDutyCycle)),
+  m_note_sensor(0)
+
+  {}
+
+void Shooter::runShooter(const double motor_output)
+{
+    m_firing_motorL.Set(-motor_output);
+    m_firing_motorR.Set(-motor_output);
+
+    runIntake(motor_output);
+}
+
+void Shooter::runIntake(const double motor_output)
+{
+    if (m_note_sensor.Get() && (-motor_output) < 0.0)
```

Comment:
Two items here:

1. Exactly what `m_note_sensor.Get()` means is a little vague. Sure, it's a Boolean, but what do true vs. false mean in this context? Is it true if we have a note, or is it false?

    Suggest creating a private function that answers this question with its name (i.e., is self-documenting). It can even be inline:

    ```cpp
    inline bool IsNoteAcquired() { return m_note_sensor.Get(); }
    ```

    I'm not wedded to the name `IsNoteAcquired()`, by the way. Perhaps there's a better one: `GotNote()`, `HasNote()`, `IsNoteInIntake()`. You might poll the team to see if they have any preferences or other suggestions.

2. Likewise, `(-motor_output) < 0.0` is not terribly clear. (_In fact, I'm still not sure I understand it, as my be evident in my suggestion that follows._) You might take a similar approach as with the note sensor:

    ```cpp
    inline IsIntakeRunning(const double motor_output) { return (-motor_output) < 0.0; }
    ```

Again, `IsIntakeRunning` may not be the best name (_or even a good one_). I.e., it may not accurately convey what the motor_output check means. But assuming for the moment that it is, notice how much more clearly the code reads if you implement the two suggestions above:

```cpp
if (IsNoteAcquired() && IsIntakeRunning(motor_output)) {
```

The code is explicit rather than implicit. It spells out what the conditions are rather than making the reader work the logic to figure out what practical conditions the code getting around to.